### PR TITLE
[[ Docs ]] Update Extending LiveCode guide

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: Documentation/guides/Extending LiveCode.md [hunk 1]
```diff
@@ -154,9 +155,9 @@ com.livecode.unittest|Default|Contains syntax for unit testing LiveCode Builder
 
 > **Warning!** Module names are subject to change.
 
-The new LiveCode dictionary has a full list of all available syntax as well as the module 
-each belongs to. As a general rule we recommend importing all three optional module 
-whenever developing widgets.
+The LiveCode dictionary has a full list of all available syntax as 
+well as the module each belongs to. As a general rule we recommend 
+importing all three optional module whenever developing widgets.
```
Comment (BerndN) [inline Documentation/guides/Extending LiveCode.md]: module -> modules

File: Documentation/guides/Extending LiveCode.md [hunk 2]
```diff
@@ -520,50 +1457,53 @@ They are installed and viewable through the LiveCode Dictionary stack.
...
-For a full list of acceptable documentation elements please see the LiveCode Documentation
-Format guide.
+For a full list of acceptable documentation elements please see the 
+LiveCode Documentation Format guide.
 
-> **Important:** Documentation is only added to the LiveCode dictionary when a full 
-extension package is installed. Currently, this is only possible through the extension
-builder.
+> **Important:** Documentation is only added to the LiveCode dictionary 
+> an extension is installed in the IDE.
```
Comment (BerndN) [inline Documentation/guides/Extending LiveCode.md]: missing when?

File: Documentation/guides/Extending LiveCode.md [hunk 3]
```diff
@@ -487,19 +688,1248 @@ The rectangle of each child widget is controlled using the following syntax:
...
+>**Note:** On Windows, it is also necessary to find out the calling 
+> convention of the function. In the majority of cases if binding to 
+> functions in the Windows API, the calling convention will be 
+> `stdcall`.
+
+The declared types of the foreign handler must match the signature, so
+that the FFI call can be computed correctly. For example, to bind to the 
+`atof` C standard library function that converts a C-string to a double,
+
+	double atof(const char *str)
```
Comment (livecodepanos) [inline Documentation/guides/Extending LiveCode.md:728]: I think the `*` here has caused the text after that to be displayed in italics, until the next `*`

Comment (livecodeali) [reply Documentation/guides/Extending LiveCode.md:728]: Yeah, I think it's just a problem with the diff viewer - if you look at it here: https://github.com/livecode/livecode-ide/blob/1666cc79a403c91c8642fdaa4f527eb48f0533e7/Documentation/guides/Extending%20LiveCode.md
it doesn't have that effect

Comment (livecodeali) [reply Documentation/guides/Extending LiveCode.md:728]: (since it is in a code block it won't apply markup)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok 3ccfa66

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 3ccfa660f99f29eb433a7e367b07d8ba64fea5fb

**Pull request merged**